common/av_common.c

```diff
@@ -80,12 +80,16 @@ AVCodecParameters *mp_codec_params_to_av(struct mp_codec_params *c)
     avp->codec_id = mp_codec_to_av_codec_id(c->codec);
     avp->codec_tag = c->codec_tag;
     if (c->extradata_size) {
-        avp->extradata =
-            av_mallocz(c->extradata_size + AV_INPUT_BUFFER_PADDING_SIZE);
+        // The internal matroska demuxer keeps the flac header - lavf doesn't
+        int skip = strncmp(c->extradata, "fLaC\0\0\0\0", 8) ? 0 : 8;
```

Comment:
Checking the codec_id/tag in addition would be safer and a good idea here IMO.

Also needs to be `memcmp`.

Reply:
Yeah this was bad and it can go think about itself in the void. (The `\0\0\0\0` was also broken (and added after initial tests): the expected last byte is 0x22 (34) for the STREAMINFO block size.)